### PR TITLE
removing rmbProxy endpoints, RMB, Ygg, and Redis

### PR DESCRIPTION
**Pull request by sameh-farouk**
### Description

as a part of switching to the new RMB implementation, we are refactoring the grid-proxy server to remove the rmb-proxy functionality and any related services such RMB, Ygg and Redis.

### Changes

- [x] Remove rmbproxy endpoints
- [x] Remove rmb, ygg, and redis
- [x] Remove all obsolete configuration
- [x] Update chart, remove RMB Kubernetes service 
- [x] Update swagger docs
- [x] Update Readme

**Removed Services:**
- RMB
- Ygg
- Redis

**Removed options:**
- redis
- substrate
- rmbTimeoutSeconds

**Removed Env Vars:**
- MNEMONICS
- SUBSTRATE
- REDIS
- RMB_TIMEOUT

**Removed Endpoints:**
/twin/*

### Related Issues

https://github.com/threefoldtech/tfgridclient_proxy/issues/291

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings

### CODE REVIEW

**Review by xmonader — CHANGES_REQUESTED** [review]
please restore everything related to tls/certs

**Comment by sameh-farouk** [comment]
> please restore everything related to tls/certs

@xmonader thanks for your review, requested changes were done in 0474a2a and 3efaf20

however I have these notes, first, I removed this code because
1. It was implemented mainly in the rmbproxy pkg, I thought that maybe was planned for something RMB related, I didn't see it used anywhere but maybe I'm wrong.
2. I understand The helm chart uses Kubernetes certificate management controller, not depend on that code.  
3. I have an impression that the code that serves TLS is not complete, also given that everywhere we start the grid proxy we pass `--no-cert` wasn't sure that it is still relevant.

now I have a question, why do we need this anyway? regarding any future docker-compose projects wouldn't be better to configure an Nginx container to do TLS Offloading instead or use our gateway?

File: Dockerfile
Comment (xmonader) [inline Dockerfile]: let's user a better name than server

File: README.md
Comment (xmonader) [inline README.md]: The compiler shouldn't be a requirement, always link to the release binaries on github, you probably need a separate developer doc e.g wiki/devsetup.md

File: README.md
Comment (xmonader) [inline README.md]: Only the dsn/connection string correct?

File: README.md
Comment (xmonader) [inline README.md:52]: Should be documented before development run

File: README.md
Comment (xmonader) [inline README.md]: Do we still need the msgbusd service? its purpose was proxying messages to zos nodes, which isn't the case anymore, all of our clients can communicate with nodes through relays

File: README.md
Comment (xmonader) [inline README.md]: Add example links e.g let's encrypt production and staging for testing.

File: README.md
Comment (xmonader) [inline README.md]: Do we need cap NET_ADMIN? we don't do any trickery with the network interfaces.

File: docs/swagger.json
Comment (xmonader) [inline docs/swagger.json]: why did we remove the healthcheck endpoint?

File: docs/swagger.yaml
Comment (xmonader) [inline docs/swagger.yaml]: Why did we remove the healthcheck endpoint?

File: charts/gridproxy/README.md
Comment (xmonader) [inline charts/gridproxy/README.md]: let's not use 3botmain let's make it gridproxy.dev.grid.tf

File: charts/gridproxy/README.md
Comment (xmonader) [inline charts/gridproxy/README.md]: What is EXPLORER_URL?

File: charts/gridproxy/values.yaml
Comment (xmonader) [inline charts/gridproxy/values.yaml]: What is that used for? also the docs says EXPLORER_URL not EXPLORER

File: cmds/proxy_server/main.go
Comment (xmonader) [inline cmds/proxy_server/main.go]: move the default to the production one please

**Review by xmonader — CHANGES_REQUESTED** [review]
Add few comments please check

**Comment by xmonader** [comment]
> > please restore everything related to tls/certs
> 
> @xmonader thanks for your review, requested changes were done in [0474a2a](https://github.com/threefoldtech/tfgridclient_proxy/commit/0474a2aa35b1290ffbdfd58b75f70a67a43118fc) and [3efaf20](https://github.com/threefoldtech/tfgridclient_proxy/commit/3efaf20b57477e400b59b9727368108d4dd49b42)
> 
> however I have these notes, first, I removed this code because
> 
> 1. It was implemented mainly in the rmbproxy pkg, I thought that maybe was planned for something RMB related, I didn't see it used anywhere but maybe I'm wrong.

It was implemented for the gridproxy itself, not related to the rmb proxying features. In general given that code for acme client is already solid in go (thanks to many server applications e.g caddy) we can reuse it in our binaries to depend on less and less, so no need to drag in more components e.g caddy or nginx


> 2. I understand The helm chart uses Kubernetes certificate management controller, not depend on that code.

It's true that gridproxy runs on kubernetes -for the time being- which wasn't the case in the beginning where we managed the proxy machines in regular VMs setups, and now we are moving away from that again to docker-compose based setups

> 3. I have an impression that the code that serves TLS is not complete, also given that everywhere we start the grid proxy we pass `--no-cert` wasn't sure that it is still relevant.
> 

It's complete and working, we are only support http-01 flow, also the usage of --no-cert is already to serve use cases where the operator deploys it when kubernetes infrastructure or with a sidecar for TLS.

> now I have a question, why do we need this anyway? regarding any future docker-compose projects wouldn't be better to configure an Nginx container to do TLS Offloading instead or use our gateway?

You can have different kinds of setups e.g on a VM, with container(s), within kubernetes. IMHO, that option with so little code, enabled variety of options for deployment, That's why I wouldn't deleted it

File: Dockerfile
Comment (sameh-farouk) [reply Dockerfile]: @xmonader any suggestions? we didn't yet agree on a new name for the stripped-down version of gridproxy so I'm not sure here.

File: docs/swagger.json
Comment (sameh-farouk) [reply docs/swagger.json]: it was a part of the rmbproxy server. restored now.

**Comment by sameh-farouk** [comment]
> Add few comments please check

@xmonader all non-doc related issues were resolved (except for the server binary name waiting for your reply).
Do you mind handling all the docs-related issues in a separate PR where docs can be focused on and revised? it would be easier also for reviewers. if so I can open an issue for the docs, link it on a new PR and work on it.

File: charts/gridproxy/values.yaml
Comment (sameh-farouk) [reply charts/gridproxy/values.yaml]: this is not used /not exist. I removed that key from the values.yaml

File: charts/gridproxy/README.md
Comment (sameh-farouk) [reply charts/gridproxy/README.md]: not used anywhere on the server or the image. should be removed, also SERVER_IP is not used, should be  SERVER_PORT. I think the docs need a full revision.

File: Dockerfile
Comment (xmonader) [reply Dockerfile]: can be gridproxy or gridrest for now

**Comment by xmonader** [comment]
Let's address the minor docs comments in this PR and do your full review in another issue / PR please

**Comment by sameh-farouk** [comment]
> Let's address the minor docs comments in this PR and do your full review in another issue / PR please

@xmonader done in f4854e5d4f007, please check.

**Comment by xmonader** [comment]
Let's update the branch before merging

**Comment by sameh-farouk** [comment]
> Let's update the branch before merging

@xmonader branch updated

**Review by xmonader — APPROVED** [review]

**Pull request merged**
